packages/components/src/components/modal/modal.jsx

```diff
@@ -68,16 +68,23 @@ const ModalElement = ({
     useOnClickOutside(wrapper_ref, closeModal, validateClickOutside);
 
     React.useEffect(() => {
-        el_ref.current.classList.add('dc-modal');
-        modal_root_ref.current.appendChild(el_ref.current);
-        if (typeof onMount === 'function') onMount();
-
+        mountModal();
         return () => {
-            modal_root_ref.current.removeChild(el_ref.current);
-            if (typeof onUnmount === 'function') onUnmount();
+            unMountModal();
         };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This is a good example where it'd be better to put `onMount` and `onUnmount` in the dependency array rather than ignoring the `exhaustive-deps` rule.

Reply:
I unresolved this as the dependencies are missing.